src/routes/Blog/components/BlogPage.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import { Redirect } from 'react-router-dom';
+import { Helmet } from 'react-helmet';
+import Post from './Post/Post';
+
+function BlogPage() {
+  const [posts, setPosts] = useState([]);
+  const [loaded, setLoaded] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    fetch(`${process.env.REACT_APP_BLOG_POST_URL}index.json`)
+      .then((res) => res.json())
+      .then((data) => {
+        setLoaded(true);
```

Comment:
In the future we may want to combine these into a single piece of state and/or use useReducer so that multiple changes aren't happening. I believe that in this case react will combine these updates into a batch, but not positive. Would be interesting to look into

Reply:
Ok. Curious if you have any example of how to implement the useReduce hook. I was thinking about how to combine reducers and if we have, in this case, some kind of store. And also how to structure the files.

Reply:
This [NotificationsProvider on the next branch](https://github.com/CV19Assist/app/blob/next/src/modules/notification/NotificationsProvider.js) uses `useReducer` with a reduce that lives [in `reducer.js` in that same module folder](https://github.com/CV19Assist/app/blob/next/src/modules/notification/reducer.js).

I have seen modules/ be a nice way to organize all of the logic together, but still have it at a level where it can be shared. That said - since this is component level, it may be best to start with all of the code in the component file. As it grows we can make a `useBlogPage` custom hook and move that to it's own file

Reply:
Perfect. Thanks¡¡